projects/fal/src/fal/api.py

```diff
@@ -824,7 +825,14 @@ async def not_found_exception_handler(request: Request, exc: HTTPException):
             # Rewrite the message to include the path that was not found.
             # This is supposed to make it easier to understand to the user
             # that the error comes from the app and not our platform.
-            return HTTPException(404, f"Path {request.url.path} not found")
+            try:
+                if exc.detail == "Not Found":
+                    return JSONResponse(
+                        {"detail": f"Path {request.url.path} not found"}, 404
+                    )
+            except Exception:
+                # If it's not a generic 404, just return the original message.
+                return JSONResponse({"detail": exc.detail}, 404)
```

Comment:
why is there a try/except here? why not just do `if exc.detail == "not found": return x else: return y`?

Reply:
wanted to be extra careful of exc.detail maybe not being an attribute? but it _should_ be